hera_pspec/uvpspec.py

```diff
@@ -179,6 +198,369 @@ def get_cov(self, key, omit_flags=False):
+            raise AttributeError("No covariance array has been calculated.")
+    '''
+
+    def get_cov_q(self, key, type="real", omit_flags=False):
+        """
+        Slice into covariance-q array with a specified data key in the format
+        (spw, ((ant1, ant2),(ant3, ant4)), pol)
+
+        or
+
+        (spw, blpair-integer, pol)
+
+        where spw is the spectral window integer, ant1 etc. arge integers,
+        and pol is either a polarization string (ex. 'XX') or integer (ex. -5).
+
+        Parameters
+        ----------
+        key: tuple
+            Baseline-pair key
+
+        type: str
+            "real" or "imag"    
+
+        omit_flags : bool, optional
+            If True, remove time integrations (or spectra) that
+            came from visibility data that were completely flagged
+            across the spectral window (i.e. integration == 0).
+
+        Returns
+        -------
+        data : complex ndarray
+            Shape (Ntimes, Ndlys, Ndlys,)
+        """
+        spw, blpairts, pol = self.key_to_indices(key, omit_flags=omit_flags)
+        # Need to deal with folded data!
+        # if data has been folded, return only positive delays
+        if type == "real":
+            if hasattr(self,'cov_array_q_real'):
+                if self.folded:
+                    Ndlys = self.data_array[spw].shape[1]
+                    result_list = []
+                    for tind in blpairts:
+                        result = odict()
+                        for key in self.cov_array_q_real[spw][tind, pol].keys():
+                            result[key] = self.cov_array_q_real[spw][tind, pol][key][Ndlys//2+1:,Ndlys//2+1:]
+                        result_list.append(result)
+                    result_list = np.array(result_list)
+                    return result_list
+                else:
+                    return self.cov_array_q_real[spw][blpairts, pol]
+            else:
+                raise AttributeError("No variance array has been calculated.")
+
+        elif type == "imag":
+            if hasattr(self,'cov_array_q_imag'):
+                if self.folded:
+                    Ndlys = self.data_array[spw].shape[1]
+                    result_list = []
+                    for tind in blpairts:
+                        result = odict()
+                        for key in self.cov_array_q_imag[spw][tind, pol].keys():
+                            result[key] = self.cov_array_q_imag[spw][tind, pol][key][Ndlys//2+1:,Ndlys//2+1:]
+                        result_list.append(result)
+                    result_list = np.array(result_list)
+                    return result_list
+                else:
+                    return self.cov_array_q_imag[spw][blpairts, pol]
+            else:
+                raise AttributeError("No covariance array has been calculated.")
+        else:
+            raise ValueError("No types besides real and imag.")
+
+    def get_cov(self, key, type="real", omit_flags=False):
```

Comment:
let's avoid using special variables like `type`. perhaps `dtype` is a better kwarg name

Reply:
Good suggestions!